File: Plugin/src/main/java/dev/rosewood/roseskyblock/island/Island.java
package dev.rosewood.roseskyblock.island;

import dev.rosewood.roseskyblock.util.SkyblockUtil;
import dev.rosewood.roseskyblock.world.IslandWorld;
import org.bukkit.Location;

public record Island(
        IslandGroup islandGroup,
        int islandId,
        IslandWorld world,
        Location spawnLocation
) {

    public Location getCenter() {
        return SkyblockUtil.getNextIslandLocation(this.islandGroup.locationId(), this.world);
    }
}

File: Plugin/src/main/kotlin/dev/rosewood/roseskyblock/island/Island.kt
package dev.rosewood.roseskyblock.island

import dev.rosewood.roseskyblock.util.getNextIslandLocation
import dev.rosewood.roseskyblock.world.IslandWorld
import org.bukkit.Location
import java.util.UUID

data class Island(
    val islandGroup: IslandGroup,
    val islandId: UUID,
    val world: IslandWorld,
    val spawnLocation: Location,
//    var settings: IslandSettings
) {

    @Suppress("unused") // we will use this later
    val center: Location by lazy { getNextIslandLocation(this.islandGroup.locationId, this.world) }
}
